### PR TITLE
Best Practices for Jenkins resource allocation

### PR DESCRIPTION
**Pull request by Markieta**
Updating `resource-tuning-recommendations.md` with recommendations to reduce over-allocation of cluster resources due to multiple Jenkins instances in `tools` namespaces.

Some of the content appended to this doc (see everything after **Advanced Jenkins Resource Tuning**) is less dev and more ops-focused. Let me know if that belongs somewhere else.

### CODE REVIEW

File: docs/resource-tuning-recommendations.md [hunk 1]
```diff
@@ -100,56 +105,62 @@ NAME              CPU(cores)   MEMORY(bytes)
...
+
+```bash
 oc get quota compute-long-running-quota -o=custom-columns=Requests:.status.used."requests\.cpu"
 ```
 
-Example output of the above, the `m` at the end again means millicores, so dividing the number by 1000 tells us the current project per this example has a total alloted CPU Requests value of 14.5 CPU cores.
+Example output of the above, the `m` at the end again means millicores, so dividing the number by 1000 tells us the current project per this example has a total allotted CPU Requests value of 14.5 CPU cores.
 
-```
+```bash
```
Comment (StevenBarre) [inline docs/resource-tuning-recommendations.md:117]:
```suggestion
```console
```

File: docs/resource-tuning-recommendations.md [hunk 2]
```diff
@@ -100,56 +105,62 @@ NAME              CPU(cores)   MEMORY(bytes)
...
 ## Jenkins Resource Configuration Recommendations
-Tuning the resources of Jenkins deployments can have a large affect on the available resources of the platform. As of the this writing, Jenkins accounts for the largest user of CPU Requests and Limits on the platform. Recent analysis has indicated:
+
+Tuning the resources of Jenkins deployments can have a large effect on the available resources of the platform. As of writing, Jenkins accounts for the largest user of CPU Requests and Limits on the platform. Recent analysis has indicated:
+
 - **15-25% of CPU Requests** on the platform are related to Jenkins
-- **7% of the CPU Requests** are actually used, on average, over a 1 Day period
-- **10% or More CPU Requests** for the overall platform can be saved by tuning Jenkins resources 
+- **7% of the CPU Requests** are actually used, on average, over 1 day
+- **10% or More CPU Requests** for the overall platform can be saved by tuning Jenkins resources
```
Comment (StevenBarre) [inline docs/resource-tuning-recommendations.md:128]: Are these numbers up to date?

File: docs/resource-tuning-recommendations.md [hunk 3]
```diff
@@ -165,14 +176,128 @@ A test was performed to collect the startup time of Jenkins under various resour
...
+
+Consider monitoring the upper and lower bounds of CPU and memory usage of Jenkins instances over time. When idle, it has been observed that Jenkins uses under `5m` of CPU and about `650Mi` of memory. As per the **General Guidelines** above, "set requests to the *minimum* of what your application needs." It is ideal to reserve resources conservatively (especially for workloads that are often idle), and leverage resource limits to burst when active.
+
+![Jenkins CPU usage](assets/../images/jenkins-cpu-usage.png)
+
+Also, consider other workloads you may need to run in the tools namespace when accounting for requests/limits allocation to be within the allotted maximums.
+
+## Tools Namespaces Resource Quota Recommendations
+
+Every product in a cluster is provided a GUID and a namespace for each environment (i.e., dev, test, prod). These products also have a **tools** namespace defined as `<guid>-tools`, where tooling such as Jenkins are deployed.
```
Comment (StevenBarre) [inline docs/resource-tuning-recommendations.md:201]:
```suggestion
Every product in a cluster is provided a licence plate and a namespace for each environment (i.e., dev, test, prod). These products also have a **tools** namespace defined as `<license>-tools`, where tooling such as Jenkins are deployed.
```

I think in most places we refer to this bit as the license plate, not GUID. @mitovskaol can you confirm?

File: docs/resource-tuning-recommendations.md [hunk 4]
```diff
@@ -165,14 +176,128 @@ A test was performed to collect the startup time of Jenkins under various resour
...
+
+From the OpenShift web console, in the **Administrator** perspective, proceed to **Administration** > **ResourceQuotas** and select the appropriate `ResourceQuota` (i.e., `compute-long-running-quota`). Here is an example:
+
+![tools example compute-long-running-quota](images/tools-example-compute-long-running-quota.png)
+
+#### Viewing Quota Usage (CLI)
+
+To describe a specific quota, use the `oc` tool:
+
+```bash
```
Comment (StevenBarre) [inline docs/resource-tuning-recommendations.md:265]:
```suggestion
```console
```

**Review by StevenBarre — COMMENTED** [review]
Should we also test using a burstable pod for Jenkins?

File: docs/resource-tuning-recommendations.md (hunk 2 above)
Comment (Markieta) [reply docs/resource-tuning-recommendations.md:128]: No, this is no longer accurate.

**Comment by Markieta** [comment]
> Should we also test using a burstable pod for Jenkins?

Yes, this is covered in the **Recommended Configuration** and **Advanced Jenkins Resource Tuning** sections below.

When setting any CPU/memory requests and higher limits, the Jenkins pods will be of QoS class `Burstable`.

**Comment by StevenBarre** [comment]
> > Should we also test using a burstable pod for Jenkins?
> 
> Yes, this is covered in the **Recommended Configuration** and **Advanced Jenkins Resource Tuning** sections below.
> 
> When setting any CPU/memory requests and higher limits, the Jenkins pods will be of QoS class `Burstable`.

Sorry I meant `BestEffort`. ie when the request and limits are set to zero.

**Comment by Markieta** [comment]
`Burstable` would be more recommended to avoid pod eviction if nodes run out of memory often. However, if Jenkins instances can be treated more ephemerally and we're not worried about this as much, we can certainly recommend `BestEffort`, though the `LimitRange` will provide those default values and should be they're right-sized for Jenkins.

File: docs/resource-tuning-recommendations.md (hunk 3 above)
Comment (mitovskaol) [reply docs/resource-tuning-recommendations.md:201]: Confirming that the correct way to refer to the project identifier must be a license plate.

**Review by StevenBarre — APPROVED** [review]

**Review by mitovskaol — APPROVED** [review]

**Pull request merged**